report/node_set.go

```diff
@@ -1,94 +1,197 @@
 package report
 
 import (
+	"bytes"
+	"encoding/gob"
+	"encoding/json"
+	"fmt"
+	"reflect"
 	"sort"
+
+	"github.com/mndrix/ps"
 )
 
-// NodeSet is a sorted set of nodes keyed on (Topology, ID). Clients must use
+// NodeSet is a set of nodes keyed on (Topology, ID). Clients must use
 // the Add method to add nodes
-type NodeSet []Node
+type NodeSet struct {
+	psMap ps.Map
+}
```

Comment:
It is used for children. But unlike parents, we summarise children, so we have to use the aggregated nodes. This way we store the nodes as we are "rolling up" the aggregations.